### PR TITLE
MOTECH-2785: Fixed unwanted blocking of the ui in MDS

### PR DESCRIPTION
**Pull request by MichalKus**
Fixed unwanted blocking of ui
Fixed modal which didn`t close when we go back to previous site

### CODE REVIEW

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

File: platform/mds/mds-web/src/main/resources/webapp/js/controllers.js [hunk 1]
```diff
@@ -2047,9 +2047,19 @@
             var exists;
 
             if ($scope.advancedSettings !== null && $scope.lookup !== undefined && $scope.lookup.lookupName !== undefined) {
```
Comment (wstrzelczyk) [inline platform/mds/mds-web/src/main/resources/webapp/js/controllers.js]: @sebbrudzinski We really need to have LoadingModal in this place?

**Comment by jredlarski** [comment]
test this please

File: platform/mds/mds-web/src/main/resources/webapp/js/controllers.js (hunk 1 above)
Comment (jredlarski) [inline platform/mds/mds-web/src/main/resources/webapp/js/controllers.js]: @sebbrudzinski @MichalKus I believe that we can get rid of it here. If so, I think the timeout should be removed as well and I would left just the validation of the lookup name

**Comment by jredlarski** [comment]
test this please

**Comment by jredlarski** [comment]
@kmadej Looks good to me

File: platform/mds/mds-web/src/main/resources/webapp/js/controllers.js [hunk 2]
```diff
@@ -2047,9 +2047,15 @@
...
-                LoadingModal.open();
                 $scope.validateLookupName($scope.lookup.lookupName);
-                LoadingModal.close();
+            }
+        });
+
+
+        $rootScope.$on('$stateChangeStart', function(event, toState, toParams, fromState, fromParams) {
```
Comment (wstrzelczyk) [inline platform/mds/mds-web/src/main/resources/webapp/js/controllers.js]: @jredlarski @MichalKus Do we really need catch every change of state in this place?

Comment (jredlarski) [inline platform/mds/mds-web/src/main/resources/webapp/js/controllers.js]: @wstrzelczyk @MichalKus We need to catch the change of state from `mds.schemaEditor` when `#advancedObjectSettingsModal` is open to force it to be closed on such state change. I can see that we catch the change of state only from `mds.schemaEditor` with either modal being open or closed

**Comment by jredlarski** [comment]
@MichalKus Instead of using method which is listening for the change of state, let's block the ability to click outside the advanced settings modal.

**Comment by jredlarski** [comment]
test this please

File: platform/mds/mds-web/src/main/resources/webapp/js/controllers.js [hunk 3]
```diff
@@ -633,7 +633,7 @@
     /**
     * The MdsSchemaEditorCtrl controller is used on the 'Schema Editor' view.
     */
-    controllers.controller('MdsSchemaEditorCtrl', function ($scope, $timeout, $http, Entities, MDSUsers, Permissions, MDSUtils, Locale, ModalFactory, LoadingModal) {
+    controllers.controller('MdsSchemaEditorCtrl', function ($scope, $timeout, $http, Entities, MDSUsers, Permissions, MDSUtils, Locale, ModalFactory, LoadingModal, $rootScope) {
```
Comment (jredlarski) [inline platform/mds/mds-web/src/main/resources/webapp/js/controllers.js]: @MichalKus Please remove `$rootScope` from this controller if it is not being used anywhere inside it

**Comment by jredlarski** [comment]
test this please

**Comment by jredlarski** [comment]
@kmadej Looks good to me

**Pull request merged**